### PR TITLE
321 [ReaderTest] testReadDouble

### LINKED ISSUE #321: [ReaderTest] testReadDouble

### PR DESCRIPTION
**Pull request by Anastasiia3795**

### CODE REVIEW

**Comment by Anastasiia3795** [comment]
Не впевнена, що то правильно, але я ще на невалідні значення перевіряю

File: src/test/java/org/codewars/utils/ReaderTest.java [hunk 1]
```diff
@@ -14,8 +20,28 @@ public void testReadInt() {
...
+    }
+    @Test(dataProvider = "readDoubleTestData")
+    public void testReadDouble(Scanner scanner, double expected) {
+        Reader reader = new Reader(scanner);
+        try {
+            double actual = reader.readDouble();
+            assertEquals(actual, expected);
+        } catch (NoSuchElementException e) {
+            assertTrue(Double.isNaN(expected));
+        }
```
Comment (mosvvy) [inline src/test/java/org/codewars/utils/ReaderTest.java:44]: mb better would be to check on few inputs? I mean smth like `{new Scanner("sda\nadwqwdcsd sdqwqwwqe12qwdc xcsd\n20,1"), 20.1f}` (it works on readFloat).

Comment (Anastasiia3795) [reply src/test/java/org/codewars/utils/ReaderTest.java]: I've added a new input

File: src/test/java/org/codewars/utils/ReaderTest.java [hunk 2]
```diff
@@ -147,43 +160,10 @@ public void testReadArrDouble() {
...
-
-    @Test(dataProvider = "testReadArrLongDataInvalid")
-    public void testReadArrLongInvalid(String str, long[] expectedValue, String expectedOut) {
-        System.setIn(new ByteArrayInputStream(str.getBytes()));
-        OutputStream out = new ByteArrayOutputStream();
-        System.setOut(new PrintStream(out));
-        Reader reader = new Reader();
-        long[] actualValue = reader.readArrLong();
-        Assert.assertEquals(out.toString().replace("\r", ""), expectedOut);
-        Assert.assertEquals(actualValue, expectedValue);
```
Comment (mosvvy) [inline src/test/java/org/codewars/utils/ReaderTest.java]: You're removing testing readArrLong

Comment (Anastasiia3795) [reply src/test/java/org/codewars/utils/ReaderTest.java]: I've added it back in the last commit

File: src/main/java/org/codewars/utils/Reader.java [hunk 3]
```diff
@@ -12,6 +12,10 @@ public Reader() {
         scanner = new Scanner(System.in);
     }
 
+    public Reader(Scanner scan) {
+        scanner = scan;
+    }
+
```
Comment (mosvvy) [inline src/main/java/org/codewars/utils/Reader.java:18]: It's needn't

Comment (Anastasiia3795) [reply src/main/java/org/codewars/utils/Reader.java]: Have changed

**Review by mosvvy — APPROVED** [review]

**Review by andriyav — APPROVED** [review]

**Review by lhalam — APPROVED** [review]

**Pull request merged**